api.go

```diff
@@ -275,17 +266,28 @@ func (api *API) addResource(prototype jsonapi.MarshalIdentifier, source CRUD) *r
 		}
 	})
 
-	api.router.Handle("GET", baseURL+"/:id", func(w http.ResponseWriter, r *http.Request, params map[string]string) {
-		info := requestInfo(r, api)
-		c := api.contextPool.Get().(APIContexter)
-		c.Reset()
-		api.middlewareChain(c, w, r)
-		err := res.handleRead(c, w, r, params, *info)
-		api.contextPool.Put(c)
-		if err != nil {
-			api.handleError(err, w, r)
-		}
-	})
+	if _, ok := source.(CRUDFindOne); ok {
+		api.router.Handle("OPTIONS", baseURL+"/:id", func(w http.ResponseWriter, r *http.Request, _ map[string]string) {
+			c := api.contextPool.Get().(APIContexter)
+			c.Reset()
+			api.middlewareChain(c, w, r)
+			w.Header().Set("Allow", "GET,PATCH,DELETE,OPTIONS")
```

Comment:
It sounds logical and I'm working on it. Kind of hard not to duplicate the code for this one and the `"GET,PATCH,DELETE,OPTIONS"` a few lines below.